src/field_theory/mv_polynomial.lean

```diff
@@ -6,13 +6,105 @@ Author: Johannes Hölzl
 Multivariate functions of the form `α^n → α` are isomorphic to multivariate polynomials in
 `n` variables.
 -/
-import linear_algebra.finsupp field_theory.finite
+import linear_algebra.finsupp_vector_space field_theory.finite data.mv_polynomial
 noncomputable theory
 
 local attribute [instance, priority 0] classical.prop_decidable
 
 open lattice set linear_map submodule
 
+namespace mv_polynomial
+universes u v
+variables {σ : Type u} {α : Type v} [decidable_eq σ]
+
+instance [discrete_field α] : vector_space α (mv_polynomial σ α) :=
+finsupp.vector_space _ _
+
+section
+variables (σ α) [discrete_field α] (m : ℕ)
+def restrict_total_degree : submodule α (mv_polynomial σ α) :=
+finsupp.supported _ _ {n | n.sum (λn e, e) ≤ m }
+
+lemma mem_restrict_total_degree (p : mv_polynomial σ α) :
+  p ∈ restrict_total_degree σ α m ↔ p.total_degree ≤ m :=
+begin
+  rw [total_degree, finset.sup_le_iff],
+  refl
+end
+
+end
+
+noncomputable instance decidable_restrict_degree (m : ℕ) :
+  decidable_pred (λn, n ∈ {n : σ →₀ ℕ | ∀i, n i ≤ m }) :=
+assume n, classical.prop_decidable _
```

Comment:
What's this for? The proof looks a bit suspicious. Is this being omitted because it's not important? Usually we get these proofs for free in mathematician mode without the instance; this is just precluding any other kind of proof.

Reply:
I don't know. I just moved this from `finsupp.lean`.

Reply:
I moved the instance above the lemma where it is needed and changed the proof to something that does not use `classical.prop_decidable`.